mithril-aggregator/src/services/cardano_transactions_importer.rs

```diff
@@ -0,0 +1,328 @@
+use async_trait::async_trait;
+use mithril_common::cardano_transaction_parser::TransactionParser;
+use mithril_common::entities::{CardanoTransaction, ImmutableFileNumber};
+use mithril_common::signable_builder::TransactionsImporter;
+use mithril_common::StdResult;
+use slog::{debug, Logger};
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+#[cfg(test)]
+use mockall::automock;
+
+/// Cardano transactions store
+#[cfg_attr(test, automock)]
+#[async_trait]
+pub trait TransactionStore: Send + Sync {
+    /// Get the highest known transaction beacon
+    async fn get_highest_beacon(&self) -> StdResult<Option<ImmutableFileNumber>>;
+
+    /// Get stored transactions up to the given beacon
+    async fn get_up_to(
+        &self,
+        immutable_file_number: ImmutableFileNumber,
+    ) -> StdResult<Vec<CardanoTransaction>>;
+
+    /// Store list of transactions
+    async fn store_transactions(&self, transactions: Vec<CardanoTransaction>) -> StdResult<()>;
+}
+
+/// Import and store [CardanoTransaction].
+pub struct CardanoTransactionsImporter {
+    transaction_parser: Arc<dyn TransactionParser>,
+    transaction_store: Arc<dyn TransactionStore>,
+    logger: Logger,
+    rescan_offset: Option<usize>,
+    dirpath: PathBuf,
+}
+
+impl CardanoTransactionsImporter {
+    /// Constructor
+    ///
+    /// About `rescan_offset`: if Some(x) the importer will be asked to rescan the previous 'x'
+    /// immutables starting from the highest immutable known in the store.
+    /// This is useful when one of the last immutable was not full scanned.
```

Comment:
This formulation is not correct as it would mean that with `x` equals to `0`, the highest immutable file number would be parsed again, which is not the case :thinking:

Reply:
Done ✅